frontend/public/js/presetcontroller.js

```diff
@@ -403,12 +410,14 @@ function updateTags() {
 * @param val value of the tag to add.
 */
 function newTag(val) {
```

Comment:
Is it maybe possible to add a return to this function, so that it returns true when it is correctly pushed to the tags (false otherwise)? Then you can know that it is correctly pushed when you call this method.